src/interface/swarm_container.cpp

```diff
@@ -75,15 +76,40 @@ void SwarmContainer::Remove(const std::string &label) {
 
 void SwarmContainer::SendBoundaryBuffers() {}
 
-void SwarmContainer::SetupPersistentMPI() {}
+void SwarmContainer::SetupPersistentMPI() {
+  for (auto &s : swarmVector_) {
+    s->SetupPersistentMPI();
+  }
+}
 
 bool SwarmContainer::ReceiveBoundaryBuffers() { return true; }
 
 void SwarmContainer::ReceiveAndSetBoundariesWithWait() {}
 
 void SwarmContainer::SetBoundaries() {}
 
-void SwarmContainer::StartReceiving(BoundaryCommSubset phase) {}
+TaskStatus SwarmContainer::Send(BoundaryCommSubset phase) {
```

Comment:
Is there a reason why this method cannot be const?

Reply:
Making that `const` gives me a weird error I don't understand, which makes me think it shouldn't be `const`:
```
In file included from /home/brryan/github/parthenon/src/driver/driver.hpp:30,
                 from /home/brryan/github/parthenon/src/parthenon/driver.hpp:21,
                 from /home/brryan/github/parthenon/example/particles/particles.hpp:20,
                 from /home/brryan/github/parthenon/example/particles/particles.cpp:14:
/home/brryan/github/parthenon/src/tasks/task_list.hpp: In instantiation of 'parthenon::TaskID parthenon::TaskList::AddTask(const parthenon::TaskID&, F&&, Args&& ...) [with F = parthenon::TaskStatus (parthenon::SwarmContainer::*)(parthenon::BoundaryCommSubset) const; Args = {parthenon::SwarmContainer*, parthenon::BoundaryCommSubset}]':
/home/brryan/github/parthenon/example/particles/particles.cpp:458:51:   required from here
/home/brryan/github/parthenon/src/tasks/task_list.hpp:77:22: error: must use '.*' or '->*' to call pointer-to-member function in 'func (...)', e.g. '(... ->* func) (...)'
   77 |           return func(args...);
      |                  ~~~~^~~~~~~~~
In file included from /home/brryan/software/spack/opt/spack/linux-ubuntu18.04-skylake/gcc-7.4.0/gcc-9.2.0-bo6tpsymozxrcsr2zwnwc5zkt7q4tpup/include/c++/9.2.0/functional:59,
                 from /home/brryan/github/parthenon/external/Kokkos/core/src/Kokkos_Core.hpp:97,
                 from /home/brryan/github/parthenon/external/Kokkos/algorithms/src/Kokkos_Random.hpp:48,
                 from /home/brryan/github/parthenon/example/particles/particles.hpp:18,
                 from /home/brryan/github/parthenon/example/particles/particles.cpp:14:
/home/brryan/software/spack/opt/spack/linux-ubuntu18.04-skylake/gcc-7.4.0/gcc-9.2.0-bo6tpsymozxrcsr2zwnwc5zkt7q4tpup/include/c++/9.2.0/bits/std_function.h:669:7: error: 'std::function<_Res(_ArgTypes ...)>::function(_Functor) [with _Functor = parthenon::TaskList::AddTask(const parthenon::TaskID&, F&&, Args&& ...) [with F = parthenon::TaskStatus (parthenon::SwarmContainer::*)(parthenon::BoundaryCommSubset) const; Args = {parthenon::SwarmContainer*, parthenon::BoundaryCommSubset}]::<lambda()>; <template-parameter-2-2> = void; <template-parameter-2-3> = void; _Res = parthenon::TaskStatus; _ArgTypes = {}]', declared using local type 'parthenon::TaskList::AddTask(const parthenon::TaskID&, F&&, Args&& ...) [with F = parthenon::TaskStatus (parthenon::SwarmContainer::*)(parthenon::BoundaryCommSubset) const; Args = {parthenon::SwarmContainer*, parthenon::BoundaryCommSubset}]::<lambda()>', is used but never defined [-fpermissive]
  669 |       function<_Res(_ArgTypes...)>::
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
example/particles/CMakeFiles/particles-example.dir/build.make:94: recipe for target 'example/particles/CMakeFiles/particles-example.dir/particles.cpp.o' failed
```